torch/_inductor/kernel/mm_common.py

```diff
@@ -133,6 +133,8 @@ def mm_grid(m, n, meta):
 
 def acc_type(dtype):
     if dtype in (torch.float16, torch.bfloat16):
+        if torch.backends.cuda.matmul.allow_fp16_reduced_precision_reduction:
+            return "tl.float16"
```

Comment:
I think we need to seperate out the float16 and bfloat16 cases because there is also:
```
torch.backends.cuda.matmul.allow_bf16_reduced_precision_reduction
```
in addition to this one.

Reply:
```python
cast_fp16 = dtype == torch.float16 and not torch.backends.cuda.matmul.allow_fp16_reduced_precision_reduction
cast_bf16 = dtype == torch.float16 and not torch.backends.cuda.matmul.allow_bf16_reduced_precision_reduction
if (cast_fp16 or cast_bf16):
  return "tl.float32"